test/controllerUtility/ctrl-utility.test.ts

```diff
@@ -916,520 +279,124 @@ const timestampEventsCases: timestampTestCases[] = [
-          rudderId: '<<>>2073230<<>>2564871',
-          receivedAt: '2022-12-23T00:29:10.189+05:30',
-          eventName: 'Test',
-          eventType: 'track',
-          sourceDefinitionId: '1b6gJdqOPOCadT3cddw8eidV591',
-          destinationDefinitionId: '',
-          transformationId: '',
-        },
-        destination: {
-          ID: 'string',
-          Name: 'string',
-          DestinationDefinition: {
-            ID: 'defid1',
-            Name: 'INTERCOM',
-            DisplayName: 'intercom',
-            Config: {},
-          },
-          Config: {},
-          Enabled: true,
-          WorkspaceID: 'wspId',
-          Transformations: [],
-        },
+        }),
+        metadata: metadata,
+        destination: destination,
       },
       {
-        message: {
+        message: getMessage({
           anonymousId: '2073232',
-          event: 'Test',
-          messageId: 'e3a51e9a-6313-4389-ae73-07e487c8d9d0',
-          originalTimestamp: '2022-12-23T00:29:12.117+05:30',
-          channel: 'sources',
           context: {
             mappedToDestination: true,
           },
           properties: {
             timestamp: '2023-01-13T00:29:12.117+05:30',
           },
-          sentAt: '2022-12-23T00:29:12.117+05:30',
           timestamp: '2022-11-22T00:29:10.188+05:30',
           type: 'group',
-          userId: '2564871',
-        },
-        metadata: {
-          sourceId: '27O0bmEEx3GgfmEhZHUcPwJQVWC',
-          workspaceId: '27O0bhB6p5ehfOWeeZlOSsSDTLg',
-          namespace: '',
-          instanceId: '1',
-          sourceType: 'HTTP',
-          sourceCategory: '',
-          trackingPlanId: '',
-          trackingPlanVersion: 0,
-          sourceTpConfig: {},
-          mergedTpConfig: {},
-          destinationId: '2JH9GMQf2YFJFaTw7rz1pxHAJPx',
-          jobRunId: '',
-          jobId: 1,
-          sourceBatchId: '',
-          sourceJobId: '',
-          sourceJobRunId: '',
-          sourceTaskId: '',
-          sourceTaskRunId: '',
-          recordId: {},
-          destinationType: 'WEBHOOK',
-          messageId: 'e3a51e9a-6313-4389-ae73-07e487c8d9d0',
-          oauthAccessToken: '',
-          messageIds: [],
-          rudderId: '<<>>2073230<<>>2564871',
-          receivedAt: '2022-12-23T00:29:10.189+05:30',
-          eventName: 'Test',
-          eventType: 'track',
-          sourceDefinitionId: '1b6gJdqOPOCadT3cddw8eidV591',
-          destinationDefinitionId: '',
-          transformationId: '',
-        },
-        destination: {
-          ID: 'string',
-          Name: 'string',
-          DestinationDefinition: {
-            ID: 'defid1',
-            Name: 'INTERCOM',
-            DisplayName: 'intercom',
-            Config: {},
-          },
-          Config: {},
-          Enabled: true,
-          WorkspaceID: 'wspId',
-          Transformations: [],
-        },
+        }),
+        metadata: metadata,
+        destination: destination,
       },
     ],
   },
```

Comment:
Also this functionality is applicable only for rETL connection events, can update 
describe('controller utility tests -- handleTimestampInEvents `for retl connections`',